src/main/scala/io/circe/yaml/Printer.scala

```diff
@@ -0,0 +1,101 @@
+package io.circe.yaml
+
+import java.io.StringWriter
+
+import scala.collection.JavaConverters._
+
+import Printer._
+import io.circe.Json
+import org.yaml.snakeyaml.emitter.Emitter
+import org.yaml.snakeyaml.nodes.Tag
+import org.yaml.snakeyaml.resolver.Resolver
+import org.yaml.snakeyaml.serializer.Serializer
+import org.yaml.snakeyaml.{DumperOptions, Yaml}
+
+final case class Printer(
+  dropNullKeys: Boolean = false,
+  indent: Int = 2,
+  maxScalarWidth: Int = 80,
+  splitLines: Boolean = true,
+  indicatorIndent: Int = 0,
+  tags: Map[String, String] = Map.empty,
+  defaultFlowStyle: FlowStyle = FlowStyle.Auto,
+  defaultScalarStyle: ScalarStyle = ScalarStyle.Plain,
+  lineBreak: LineBreak = LineBreak.Unix,
+  version: YamlVersion = YamlVersion.Yaml1_1
+) {
+  final def pretty(json: Json): String = {
+    val options = {
+      val options = new DumperOptions()
+      options.setIndent(indent)
+      options.setWidth(maxScalarWidth)
+      options.setSplitLines(splitLines)
+      options.setIndicatorIndent(indicatorIndent)
+      if(tags.nonEmpty) options.setTags(tags.asJava)
+      options.setDefaultFlowStyle(defaultFlowStyle match {
```

Comment:
It seems kind of odd to have all these mappings inside a definition inside a pretty-printing method.